source/postprocess/visualization/compositional_vector.cc

```diff
@@ -0,0 +1,191 @@
+            }
+            prm.leave_subsection();
+          }
+          prm.leave_subsection();
+        }
+        prm.leave_subsection();
+      }
+
+
+      template <int dim>
+      void
+      CompositionalVector<dim>::parse_parameters (ParameterHandler &prm)
+      {
+        prm.enter_subsection("Postprocess");
+        {
+          prm.enter_subsection("Visualization");
+          {
+            prm.enter_subsection("Compositional fields as vectors");
+            {
+              vector_names = Utilities::split_string_list(prm.get("Names of vectors"), ',');
+
+              const std::vector<std::string> sets_list = Utilities::split_string_list(prm.get("Names of fields"), ';');
+              for (unsigned int i=0; i<sets_list.size(); ++i)
+                {
+                  const std::vector<std::string> set = Utilities::split_string_list(sets_list[i], ',');
+                  AssertThrow(set.size() == dim,
+                              ExcMessage("Sets of compositional fields to be output as vectors must have dim components."));
+
+                  std::vector<unsigned int> set_idx;
+                  for (unsigned int j=0; j<dim; ++j)
+                    {
+                      AssertThrow(this->introspection().compositional_name_exists (set[i]),
+                                  ExcMessage("All fields in compositional vector sets must match names of compositional "
+                                             "fields as assigned in the \"Compositional fields.Names of fields\" parameter."));
+                      set_idx.push_back(this->introspection().compositional_index_for_name(set[j]));
+                    }
+                  sets.push_back(set_idx);
+                }
+
+              AssertThrow(vector_names.size() == sets.size(),
+                          ExcMessage("You must define names for each of the sets of compositional fields which will "
+                                     "be output as vectors, and the length of that list must match the number of vectors."));
+            }
+            prm.leave_subsection();
+          }
+          prm.leave_subsection();
+        }
+        prm.leave_subsection();
+      }
+    }
+  }
+}
+
+
+// explicit instantiations
+namespace aspect
+{
+  namespace Postprocess
+  {
+    namespace VisualizationPostprocessors
+    {
+      ASPECT_REGISTER_VISUALIZATION_POSTPROCESSOR(CompositionalVector,
+                                                  "compositional vector",
+                                                  "A visualization output object that outputs vectors whose "
+                                                  "components are derived from compositional fields. Input "
+                                                  "parameters for this postprocessor are defined in section "
+                                                  "Postprocess.Visualization.Compositional fields as vectors")
```

Comment:
we typically write paths of subsections with a forward slash (so "Postprocess/Visualization/Compositional fields").